packages/firestore/test/unit/remote/bloom_filter.test.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { expect } from 'chai';
+
+import { BloomFilter } from '../../../src/remote/bloom_filter';
+import { ByteString } from '../../../src/util/byte_string';
+
+import * as TEST_DATA from './bloom_filter_golden_test_data';
+
+describe('BloomFilter', () => {
+  it('can instantiate an empty bloom filter', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array(0), 0, 0);
+    expect(bloomFilter.size).to.equal(0);
+  });
+
+  it('should throw error if empty bloom filter inputs are invalid', () => {
+    expect(() => new BloomFilter(new Uint8Array(0), 1, 0)).to.throw(
+      'Invalid padding when bitmap length is 0: 1'
+    );
+    expect(() => new BloomFilter(new Uint8Array(0), 0, -1)).to.throw(
+      'Invalid hash count: -1'
+    );
+  });
+
+  it('can instantiate a non empty bloom filter', () => {
+    const bloomFilter = new BloomFilter(new Uint8Array([255, 255, 124]), 3, 13);
+    expect(bloomFilter.size).to.equal(21);
+  });
+
+  it('should throw error if padding is invalid', () => {
+    expect(() => new BloomFilter(new Uint8Array(1), -1, 1)).to.throw(
+      'Invalid padding: -1'
+    );
+    expect(() => new BloomFilter(new Uint8Array(1), 9, 1)).to.throw(
+      'Invalid padding: 9'
+    );
+  });
+
+  it('should throw error if hash count is negative', () => {
+    expect(() => new BloomFilter(new Uint8Array(1), 1, -1)).to.throw(
+      'Invalid hash count: -1'
+    );
+  });
+
+  it('should throw error if hash count is 0 for non empty bloom filter', () => {
+    expect(() => new BloomFilter(new Uint8Array(1), 1, 0)).to.throw(
+      'Invalid hash count: 0'
+    );
+  });
+
+  it('should be able to process non standard characters', () => {
```

Comment:
Make sure the non standard characters bloom filter result is the same with backend